From: Sathya Perla <sathya.perla@emulex.com>
Date: Sun, 26 Jun 2011 20:41:53 +0000
Subject: [PATCH] be2net: fix initialization of vlan_prio_bmap
References: bnc#722092
Patch-mainline: v3.0-rc4
Git-commit: 2903dd654d8788425a9523959b02933ea6555229

Initialization of this field to "all priorities" must be done before MCC queue
creation. As soon as the MCC queue is created, an event modifying this value
may be received.

Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/benet/be_main.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -2556,6 +2556,9 @@ static int be_setup(struct be_adapter *a
 	if (status != 0)
 		goto tx_qs_destroy;
 
+	/* Allow all priorities by default. A GRP5 evt may modify this */
+	adapter->vlan_prio_bmap = 0xff;
+
 	status = be_mcc_queues_create(adapter);
 	if (status != 0)
 		goto rx_qs_destroy;
@@ -3421,10 +3424,6 @@ static int __devinit be_probe(struct pci
 	}
 
 	dev_info(&pdev->dev, "%s port %d\n", nic_name(pdev), adapter->port_num);
-	/* By default all priorities are enabled.
-	 * Needed in case of no GRP5 evt support
-	 */
-	adapter->vlan_prio_bmap = 0xff;
 
 	schedule_delayed_work(&adapter->work, msecs_to_jiffies(100));
 	return 0;
